### PR TITLE
JUnit tests org.eclipse.kura.raspberrypi.sensehat

### PR DESCRIPTION
**Pull request by darionct**
Created unit tests for sensors.
Created dummy device to simulate HW devices.

KuraI2CDevice.class replaces real implementation of the same name because 
it comes first in the class path when tests are run.
Devices are simulated with dummy implementation which returns 
and stores values using the "register" variable.

Signed-off-by: Dario Novakovic <dario.novakovic@comtrade.com>

### CODE REVIEW

**Comment by cdealti** [comment]
OK to merge it.

File: kura/test/org.eclipse.kura.raspberrypi.sensehat.test/src/test/java/org/eclipse/kura/raspberrypi/sensehat/sensors/HTS221Test.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        float qt = (float) TestUtil.getFieldValue(hts221, "qt");
+        assertTrue(mt * (TEMP_H_REG << 8 | TEMP_L_REG) + qt == temperature);
+    }
+
+    /*
+     * requires #1267 issue to be fixed
+     * fixed in PR #1271
+     * 
+     * @Test
+     * public void testIsHumidityReady() {
```
Comment (cdealti) [inline kura/test/org.eclipse.kura.raspberrypi.sensehat.test/src/test/java/org/eclipse/kura/raspberrypi/sensehat/sensors/HTS221Test.java]: @darionct please enable this as PR #1271 has been already merged

**Review by cdealti — CHANGES_REQUESTED** [review]

**Comment by MMaiero** [comment]
@darionct Please perform a rebase instead of a merge when you have to align your branches. Also try to shape your commits to self-contained ones, in order that the code always builds. Thanks

**Comment by darionct** [comment]
Mateo,


I know that #1274 is a mess, but i never wanted to do it, git led me that way.


Have you got any idea how to fix this? I tried squashing but its not working because of this last "merged commit".



I understand your point, will do next time. Is there anything we can do now?



________________________________

**Comment by darionct** [comment]
Closing due to the git merging wen wrong. Replacement PR will be created.

**Pull request closed without merging**